llvm/lib/Target/DirectX/DXILFinalizeLinkage.cpp

```diff
@@ -28,6 +28,14 @@ static bool finalizeLinkage(Module &M) {
     }
   }
 
+  // Remove unused global variables.
+  for (GlobalVariable &GV : make_early_inc_range(M.globals())) {
+    if (GV.use_empty()) {
+      GV.eraseFromParent();
+      MadeChange = true;
+    }
+  }
```

Comment:
style nit: unecessary curly braces
```suggestion
  for (GlobalVariable &GV : make_early_inc_range(M.globals()))
    if (GV.use_empty()) {
      GV.eraseFromParent();
      MadeChange = true;
    }
```